### DISCUSSION TITLE
How to add binaries for an MSI file in WiX Setup project

### DISCUSSION THREAD

**Discussion by sansa20** [wixtoolset/issues#7389] [Questions]
I am trying to create an MSI file with all the binaries I have available. As of now my code for the binaries looks something like this:-
```
<Component Id="OpenTap.dll" Guid="">
				<File KeyPath="yes" Source="D:\Workspace" />
			</Component>

<Component Id="OpenTap.Package.dll" Guid="">
				<File KeyPath="yes" Source="D:\\Workspace" />
			</Component>

<Component Id="tap.exe" Guid="">
				<File KeyPath="yes" Source="D:\Workspace" />
			</Component>
```
I have over 30 dlls so I am using a separate component element for each dll. This seems a bit inefficient. Is there any way I can just use the component element once, to point towards the folder that consists all these dlls.

**Comment by robmen** [comment]
At FireGiant, we developed [an advanced harvesting technology](https://www.firegiant.com/products/wix-expansion-pack/#advanced-harvesting) that behaves exactly like that.

**Reply by sansa20** [reply]
I was looking through the above documentation and tried following the same for my project, and I understood that the advanced harvesting tool is a part of the WiX Expansion package which isn't open source. Would there be any other alternative to approach this
